beesdoo_shift/models/task.py

```diff
@@ -104,19 +117,27 @@ def _read_group_stage_id(self, ids, domain, read_group_order=None, access_rights
 
     #TODO button to replaced someone
     @api.model
-    def unsubscribe_from_today(self, worker_ids, today=None, end_date=None):
-        today = today or fields.Date.today()
-        today += ' 00:00:00'
-        if end_date:
-            end_date += ' 23:59:59'
-        # date_domain = [('worker_id', 'in', worker_ids), ('start_time', '>=', today)]
-        date_domain = [('start_time', '>=', today)]
+    def unsubscribe_from_today(self, worker_ids, today=None, end_date=None, now=None):
+        """
+            If 'now' is set (must be a datetime), start_time and end_time
+            are replaced and worker is unsuscribed from 'now'.
+        """
+        if not now:
+            today = today or fields.Datetime.now()
+            today += ' 00:00:00' # Only if today not None
+            if end_date:
+                end_date += ' 23:59:59'
+        else:
+            if len(now) != 19:
+                raise UserError (_("'Now' must be a datetime."))
+            today = now
+        date_domain = [('start_time', '>', today)]
         if end_date:
             date_domain.append(('end_time', '<=', end_date))
```

Comment:
Here you should not use `end_date` if `now` is given, or you should ensure that `end_date` is a datetime format like you do when `now` is not given.